lib/src/routing/connection_registry.rs

```diff
@@ -1,350 +1,221 @@
+                            servers
+                        }
                         Err(e) => {
-                            debug!("Failed to refresh routing table: {}", e);
-                            ttl
+                            error!("Failed to refresh routing table for database {db_name}: {e}");
+                            vec![]
                         }
-                    };
+                    }
+                } else {
+                    table.resolve()
                 }
-                // Handle forced updates
-                cmd = rx.recv() => {
-                    match cmd {
-                        Some(RegistryCommand::RefreshSingleTable((db, new_bookmarks))) => {
-                            let db_name = db.as_ref().map(|d| d.to_string()).unwrap_or_default();
-                            debug!("Forcing refresh of routing table for database: {}", db_name);
-                            bookmarks = new_bookmarks;
-                            ttl = match refresh_routing_table(&config_clone, &registry.pool_registry, provider.clone(), bookmarks.as_slice(), db).await {
-                                Ok(table) => {
-                                    registry.databases.insert(db_name.clone(), table.resolve());
-                                    // we don't want to lose the initial TTL synchronization: if the forced update is triggered,
-                                    // we derive the TTL from the initial time. Example:
-                                    // if the TTL is 60 seconds and the forced update is triggered after 10 seconds,
-                                    // we want to set the TTL to 50 seconds, so that the next update will be in 50 seconds
-                                    ttl - (now.elapsed().as_secs() % table.ttl)
-                                }
-                                Err(e) => {
-                                    error!("Failed to refresh routing table: {}", e);
-                                    // we don't want to lose the initial TTL synchronization: if the forced update is triggered,
-                                    // we derive the TTL from the initial time. Example:
-                                    // if the TTL is 60 seconds and the forced update is triggered after 10 seconds,
-                                    // we want to set the TTL to 50 seconds, so that the next update will be in 50 seconds
-                                    ttl - (now.elapsed().as_secs() % ttl)
-                                }
-                            };
-                        }
-                        Some(RegistryCommand::Stop) | None => {
-                            debug!("Stopping background updater");
-                            break;
-                        }
+            } else {
+                match self
+                    .fetch_routing_table(db.clone(), imp_user.clone(), bookmarks)
```

Comment:
`imp_user` can be used directly without a clone

```suggestion
                    .fetch_routing_table(db.clone(), imp_user, bookmarks)
```